lib/services/jira.rb

```diff
@@ -102,14 +102,16 @@ def search_issues(params)
   end
 
   def issue_type_by_id(id)
-    raise AhaService::RemoteError, "Integration has not been configured" if meta_data.projects.nil?
     project = meta_data.projects.find {|project| project[:key] == data.project }
-    raise AhaService::RemoteError, "Integration has not been configured, can't find project '#{data.project}'" if project.nil?
-    raise AhaService::RemoteError, "Integration has not been configured, project data not loaded" if meta_data.issue_type_sets.nil?
     issue_types = meta_data.issue_type_sets[project.issue_types]
     issue_type = issue_types.find {|type| type.id.to_s == id.to_s }
-    raise AhaService::RemoteError, "Integration needs to be reconfigured, issue types have changed, can't find issue type '#{id}'" if issue_type.nil?
-    issue_type
+  rescue
+    if meta_data.projects.nil? then raise AhaService::RemoteError, "Integration has not been configured" end
+    if project.nil? then raise AhaService::RemoteError, "Integration has not been configured, can't find project '#{data.project}'" end
+    if meta_data.issue_type_sets.nil? or project.issue_types.nil? or issue_types.nil? then raise AhaService::RemoteError, "Integration has not been configured, project data not loaded" end
+    raise # re-raise original error
+  else
```

Comment:
Having the `else` here seems just a bit confusing since the main deciding factor of whether it's sole execution will happen is on the other side of the `rescue`. It seems oddly distant from the `issue_type`.

It feels more natural to do something like this prior to the rescue:

```ruby
...
  issue_types = meta_data.issue_type_sets[project.issue_types]
  issue_type = issue_types.find {|type| type.id.to_s == id.to_s }

  if issue_type.nil? then raise AhaService::RemoteError, "Integration needs to be reconfigured, issue types have changed, can't find issue type '#{id}'" end
rescue
  if meta_data.projects.nil? then raise AhaService::RemoteError, "Integration has not been configured" end
...
```